src/Features/CSharp/Portable/ConvertToRawString/ConvertRegularStringToRawStringCodeRefactoringProvider.cs

```diff
@@ -251,9 +251,15 @@ private static bool HasLeadingWhitespace(VirtualCharSequence characters)
             if (kind == ConvertToRawKind.MultiLineWithoutLeadingWhitespace)
                 characters = CleanupWhitespace(characters);
 
-            return kind == ConvertToRawKind.SingleLine
-                ? ConvertToSingleLineRawString(token, characters)
-                : ConvertToMultiLineRawIndentedString(document, token, options, characters, cancellationToken);
+            if (kind == ConvertToRawKind.SingleLine)
+            {
+                return ConvertToSingleLineRawString(token, characters);
+            }
+
+            var indentationOptions = new IndentationOptions(formattingOptions);
+            var parsedDocument = await ParsedDocument.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var indentation = token.GetPreferredIndentation(parsedDocument, indentationOptions, cancellationToken);
+            return ConvertToMultiLineRawIndentedString(indentation, token, formattingOptions, characters);
```

Comment:
why were we not async before?

Reply:
General problem that this PR is fixing: core component (like indenter) was used from both command handlers (sync code) that waited on the result and also from async code paths like code fixes. Components that were semi-synchronous   (waiting for async task) were causing problems for async callers and vice versa. Making them truly sync and pushing the asynchrony up makes all callers happy.